api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/support/x509/X509Properties.java

```diff
@@ -261,7 +261,11 @@
         /**
          * Create principal by common name and EDIPI.
          */
-        CN_EDIPI
+        CN_EDIPI,
+        /**
+         * Create principal by email in subject alternative name email type.
```

Comment:
Could you explain how this particular type is different from a normal email? What is RFC822 about, briefly?

Reply:
From what I saw the other support for EMAILADDRESS in the X509SubjectPrincipalResolver was parsing an email address out of the subject field. This is pulling an e-mail address out of the SAN (subject alternative name) field. From what I can tell RFC822 is just a synonym for the format of an e-mail address but when you look at the certificate in a certificate viewer the type of the SAN is listed as "RFC822 Name". Another way to describe it would be "SAN E-mail". Of the various types of names in the SAN, this is type "1" as opposed to type "0" which is "Other Name" and that is where the user principal name is stored (used by windows when you login to a workstation with a smart card). 

I hope I didn't miss some code that is already pull this out? 

This is from the X509Certificate javadoc:
`
 GeneralName ::= CHOICE {
      otherName                       [0]     OtherName,
      rfc822Name                      [1]     IA5String,
      dNSName                         [2]     IA5String,
      x400Address                     [3]     ORAddress,
      directoryName                   [4]     Name,
      ediPartyName                    [5]     EDIPartyName,
      uniformResourceIdentifier       [6]     IA5String,
      iPAddress                       [7]     OCTET STRING,
      registeredID                    [8]     OBJECT IDENTIFIER}
 `

Reply:
Thank you. Your note above is perfect, just make sure it ends up in the (java)docs for this field and elsewhere as appropriate. (Lose the first person pronouns of course)

...and no, I am rather sure there is nothing in CAS dealing with this, save your change here.